app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -256,17 +258,22 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
         })
         return
       }
-      else -> availableSpaceCalculator.hasAvailableSpaceFor(item,
-        { downloadFile(item.book) },
-        {
-          libraryList.snack(
-            getString(R.string.download_no_space) +
-              "\n" + getString(R.string.space_available) + " " +
-              it,
-            R.string.download_change_storage,
-            ::showStorageSelectDialog
-          )
-        })
+      else -> if (sharedPreferenceUtil.showStorageOption) {
+        showStorageConfigureDialog()
+      } else {
+        availableSpaceCalculator.hasAvailableSpaceFor(item,
+          { downloadFile(item.book) },
+          {
+            libraryList.snack(
+              "${getString(R.string.download_no_space)} " +
+                "\n  ${getString(R.string.space_available)}" +
+                it,
```

Comment:
```suggestion
              "${getString(R.string.download_no_space)}\n${getString(R.string.space_available)} $it",
```
We don't use "+" for joining Strings in Kotlin, this is called _concatenation_, instead interpolation is used to make the code more clean and legible
You can learn about string interpolation from here: https://kotlinlang.org/docs/idioms.html#string-interpolation

Reply:
oh yes, I do remember doing this but when I committed this I got the error 
``Exceeded max line length`` so I had to take them to the next line which automatically added '+'
any suggestion to resolve this and use your suggested code instead?

Reply:
Cool, you can fix lint errors via hitting `Ctrl + Alt + L`, and in this case can use kotlin multiline strings:
https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#use-multiline-strings

Reply:
sure will commit these changes soon :)

Reply:
>Cool, you can fix lint errors via hitting Ctrl + Alt + L, and in this case can use kotlin multiline strings:
https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#use-multiline-strings
I tried it but using that doesn't give proper output
![image](https://user-images.githubusercontent.com/75211982/145125345-9c386998-de04-4a1d-81bd-3f4d48551921.png)

Reply:
>Cool, you can fix lint errors via hitting Ctrl + Alt + L, and in this case can use kotlin multiline strings:
https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#use-multiline-strings

@s-ayush2903 I tried it but using that doesn't give proper output
![image](https://user-images.githubusercontent.com/75211982/145125345-9c386998-de04-4a1d-81bd-3f4d48551921.png)

Reply:
You're supposed to figure that on your own, its not difficult :stuck_out_tongue: , you can try and test on scratches in android studio.
Can refer: https://www.jetbrains.com/help/idea/scratches.html & https://medium.com/@shivam.gosavi340_58315/productivity-hack-androidstudio-kotlin-scratch-file-e17cfde152c6

Reply:
okay sure :)

Reply:
Good work. Btw, can you please put a space on the rhs of colon, like it'll be easier to read if written as:
`Space available: xxx` instead `Space available:xxx`

Reply:
![image](https://user-images.githubusercontent.com/75211982/146151222-3fa4f40f-e94f-43f9-8267-b6047a2fb8ed.png)
done, thanks for suggestion that does makes it easier to read :)